src/analysis.ts

```diff
@@ -102,21 +104,38 @@ export async function analyzeFolders(options: FileAnalysisOptions): Promise<File
   });
   if (fileBundle === null) return null;
 
-  // Analyze bundle
-  const analysisResults = await analyzeBundle({
+  const config = {
     bundleHash: fileBundle.bundleHash,
     ...options.connection,
     ...options.analysisOptions,
     shard: calcHash(fileBundle.baseDir),
     ...(options.analysisContext ? { analysisContext: options.analysisContext } : {}),
-  });
+  };
+
+  let analysisResults: AnalysisResult;
+
+  // Whether this is a report/result upload operation.
+  const isReport = options.reportOptions.enabled;
+  let reportResults: ReportUploadResult | undefined;
+  if (isReport) {
```

Comment:
q: I'm double-checking - we have made absolutely sure that we do not impact existing workflows with this change?